xinference/api/restful_api.py

```diff
@@ -808,6 +828,60 @@ async def build_gradio_interface(
 
         return JSONResponse(content={"model_uid": model_uid})
 
+    async def build_gradio_images_interface(
+        self, model_uid: str, request: Request
+    ) -> JSONResponse:
+        """
+        Build a Gradio interface for image processing models.
+        """
+        payload = await request.json()
+        try:
+            body = BuildGradioImageInterfaceRequest.parse_obj(payload)
+        except ValidationError as e:
+            logger.info(e.json())
+        logger.info("[body] %s", body)
+        assert self._app is not None
+        assert body.model_type == "image"
+
+        # asyncio.Lock() behaves differently in 3.9 than 3.10+
+        # A event loop is required in 3.9 but not 3.10+
+        if sys.version_info < (3, 10):
+            try:
+                asyncio.get_event_loop()
+            except RuntimeError:
+                warnings.warn(
+                    "asyncio.Lock() requires an event loop in Python 3.9"
+                    + "a placeholder event loop has been created"
+                )
+                asyncio.set_event_loop(asyncio.new_event_loop())
+
+        from ..core.text2image_interface import Text2ImageInterface
+
+        try:
+            access_token = request.headers.get("Authorization")
+            internal_host = "localhost" if self._host == "0.0.0.0" else self._host
+            interface = Text2ImageInterface(
+                endpoint=f"http://{internal_host}:{self._port}",
+                model_uid=model_uid,
+                model_family=body.model_family,
+                model_name=body.model_name,
+                model_id=body.model_id,
+                model_revision=body.model_revision,
+                controlnet=body.controlnet,
+                access_token=access_token,
+            ).build()
+
+            gr.mount_gradio_app(self._app, interface, f"/{model_uid}")
+        except ValueError as ve:
+            logger.error(str(ve), exc_info=True)
```

Comment:
Why str(ve)?

Reply:
By passing str(ve), you're converting the exception object into a human-readable string that describes the error that occurred, allowing it to be logged.